### PR TITLE
Fixes #20273: dashboard only shows the compliance of the nodes that really answer when logging on the test platform in 7.0

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/20273

### CODE REVIEW

**Comment by ncharles** [comment]
actually, the problem sometime still happen

**Comment by ncharles** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala [hunk 1]
```diff
@@ -746,12 +746,12 @@ final case class ContextForNoAnswer(
...
         }
 
-      case NoReportInInterval(expectedConfig, _) =>
+      case NoReportInInterval(expectedConfig, expirationTime) =>
         ComplianceDebugLogger.node(nodeId).trace(s"Node didn't received reports recently, status depend of the compliance mode and previous report status")
         buildRuleNodeStatusReport(
             //these reports don't really expires - without change, it will
             //always be the same.
-            MergeInfo(nodeId, None, Some(expectedConfig.nodeConfigId), END_OF_TIME)
+            MergeInfo(nodeId, None, Some(expectedConfig.nodeConfigId), expirationTime)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala:754]: that seems to contradict the comment just above, what changed in the logic that implies that change in behaviour?

Comment (ncharles) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala:754]: ha, indeed the comment is wrong

**Comment by ncharles** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**